connect/mirror/src/main/java/org/apache/kafka/connect/mirror/DefaultTopicFilter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.mirror;
+
+import org.apache.kafka.common.Configurable;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/** Uses a whitelist and blacklist. */
+public class DefaultTopicFilter implements TopicFilter, Configurable {
+    
+    public static final String TOPICS_WHITELIST_CONFIG = "topics";
+    private static final String TOPICS_WHITELIST_DOC = "List of topics and/or regexes to replicate.";
+    public static final String TOPICS_WHITELIST_DEFAULT = ".*";
+
+    public static final String TOPICS_BLACKLIST_CONFIG = "topics.blacklist";
+    private static final String TOPICS_BLACKLIST_DOC = "List of topics and/or regexes that should not be replicated.";
+    public static final String TOPICS_BLACKLIST_DEFAULT = ".*[\\-\\.]internal, .*\\.replica, __consumer_offsets";
```

Comment:
Whether to include `__transaction_state` internal topic here?

Reply:
yeah let's blacklist anything with `__` prefix, thanks.

Reply:
On this note we probably want to document that it doesn't work for transactional topics (also as this was one of the first questions on the summit :) )

Reply:
Related to 

> doesn't work for transactional topics

I assume with the addition of record header copying, it will also only work for when `log.message.format.version` >= `0.11.0` ? 

At least, we've noticed that when clients try to use headers (or transactional producers) after broker upgrades, but before log format changes, then they are usually throwing `UnknownServerException`. 

---

One workaround I did (for an SMT) was to conditionally copy the headers to the transformed record. e.g. https://github.com/cricket007/schema-registry-transfer-smt/blob/master/src/main/java/cricket/jmoore/kafka/connect/transforms/SchemaRegistryTransfer.java#L157-L168

Reply:
@cricket007 My intention is to support 0.11.0 onwards, at least for now. We can revisit making headers optional to support older versions, but I think this probably isn't the only thing that would break before 0.11.0.